ui/backend/client_ui_sqlite.py

```diff
@@ -165,3 +172,26 @@ def get_files_summary(self, repo_url):
                 " FROM discoveries WHERE repo_url=?"
                 " GROUP BY file_name"
             ))
+
+    def update_similar_snippets(self,
```

Comment:
why do you need to redefine this (or to copy it from the other class)?

Reply:
Precisely because of the get_discoveries method that is specific to the UI classes. No other way to go than to rewrite the adapted version of update_similar_snippets in the UI classes

Reply:
Or maybe checking for length of method output directly in client.py, but doesn't seem any more efficient